### PR TITLE
Add project archetypes for data renderer projects

### PR DESCRIPTION
**Pull request by gecage952**
Added two maven archetypes for generating data element projects. The dependencies for the data renderer are pulled in by the parent pom, and the archetypes are children of that. I only did that so you would only have to change that dependency in one place if it needed to be changed, but maybe that isn't a big deal. The webform archetype is based on the basic vaadin archetype and connects to spring boot. Just like the vaadin one, you can run "mvn jetty:run" to test it without needing to do anything after generating the project.

### CODE REVIEW

**Comment by jayjaybillings** [comment]
Thanks for sharing this. Have you created an account at Eclipse.org, linked it to your GitHub account, and signed the Contributor License Agreement (electronically)?

More information here: https://github.com/eclipse/ice/blob/master/CONTRIBUTING.md

**Comment by gecage952** [comment]
I just checked and it says I have signed the agreement. I didn't sign the commit though.

**Comment by jayjaybillings** [comment]
@gecage952 Okay, thanks. Will you amend your commit please to sign it off? You can do that with

```bash
git commit --amend --signoff
git push
```

**Comment by gecage952** [comment]
Ok. On my fork, it says the commits have been signed off (accidentally did a couple times, because I wasn't sure).

**Comment by jayjaybillings** [comment]
I think the problem is that the email address you are using to sign off on the commits does not match the email address that you used with Eclipse. You need to use the same address at Eclipse.org as you do in your signoff, which should be your ornl.gov address.

**Comment by jayjaybillings** [comment]
@osbornjd As the resident record holder for most messed up PR based on Eclipse requirements, can you help @gecage952 sort out his problems please?

Since he originally started with only a single commit, a rebase with a signoff on it might be the simplest way.

**Comment by jayjaybillings** [comment]
Thanks @osbornjd. Daniel jumped in to help @gecage952 and they got it sorted.

Since the checks are passing, this PR is open for review and I encourage everyone to look at it. This will be the basis of how we generate projects and code going forward.

@gecage952 Do you have an example posted anywhere of what it produces?

File: org.eclipse.ice.archetypes/data-element-archetype/src/main/resources/archetype-resources/pom.xml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	</properties>
+
+	<dependencies>
+		<dependency>
+			<groupId>junit</groupId>
+			<artifactId>junit</artifactId>
+			<version>4.11</version>
+			<scope>test</scope>
+		</dependency>
+	</dependencies>
```
Comment (dbluhm) [inline org.eclipse.ice.archetypes/data-element-archetype/src/main/resources/archetype-resources/pom.xml:29]: If I'm looking at the right file, don't we want to include `org.eclipse.ice.dev.annotations` in the dependencies here?

File: org.eclipse.ice.archetypes/data-element-archetype/src/main/resources/archetype-resources/pom.xml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+				</plugin>
+				<!-- see http://maven.apache.org/ref/current/maven-core/default-bindings.html#Plugin_bindings_for_jar_packaging -->
+				<plugin>
+					<artifactId>maven-resources-plugin</artifactId>
+					<version>3.0.2</version>
+				</plugin>
+				<plugin>
+					<artifactId>maven-compiler-plugin</artifactId>
+					<version>3.8.0</version>
+				</plugin>
```
Comment (dbluhm) [inline org.eclipse.ice.archetypes/data-element-archetype/src/main/resources/archetype-resources/pom.xml:47]: And the annotation processor preloaded here?
```suggestion
				<plugin>
					<artifactId>maven-compiler-plugin</artifactId>
					<version>3.8.0</version>
					<configuration>
						<release>11</release>
						<annotationProcessorPaths>
							<path>
								<groupId>org.eclipse.ice.dev</groupId>
								<artifactId>org.eclipse.ice.dev.annotations</artifactId>
								<version>3.0.0-SNAPSHOT</version>
							</path>
						</annotationProcessorPaths>
					</configuration>
				</plugin>
```

File: org.eclipse.ice.archetypes/data-element-archetype/src/main/resources/archetype-resources/src/main/java/App.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+/**
+ * Hello world!
+ *
+ */
+public class App 
+{
+    public static void main( String[] args )
+    {
+        System.out.println( "Hello World!" );
```
Comment (dbluhm) [inline org.eclipse.ice.archetypes/data-element-archetype/src/main/resources/archetype-resources/src/main/java/App.java:11]: It'd be cool to have an example DataElement specification class included in the archetype. Perhaps something like:

```java
package $package

import org.eclipse.ice.dev.annotations.DataElement;
import org.eclipse.ice.dev.annotations.DataField;

@DataElement(name = "Example")
public class ExampleSpec {
	@DataField public String exampleField;
}
```

File: org.eclipse.ice.archetypes/src/main/java/org/eclipse/ice/App.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+ * Hello world!
+ *
+ */
+public class App 
+{
+    public static void main( String[] args )
+    {
+        System.out.println( "Hello World!" );
+    }
+}
```
Comment (dbluhm) [inline org.eclipse.ice.archetypes/src/main/java/org/eclipse/ice/App.java:13]: Is this file needed for the archetype? (I'm guessing no since it looks like generated content :slightly_smiling_face:)

File: org.eclipse.ice.archetypes/data-element-archetype/src/main/resources/archetype-resources/src/main/java/App.java (hunk 3 above)
Comment (dbluhm) [reply org.eclipse.ice.archetypes/data-element-archetype/src/main/resources/archetype-resources/src/main/java/App.java:11]: Oh, maybe even something showing some interaction with DataElements in the generic `main` method would be kind of nice. Thoughts @jayjaybillings?

File: org.eclipse.ice.archetypes/data-element-archetype/src/main/resources/archetype-resources/pom.xml (hunk 1 above)
Comment (dbluhm) [reply org.eclipse.ice.archetypes/data-element-archetype/src/main/resources/archetype-resources/pom.xml:29]: Just saw your comments in your OP -- seems I might be off. Which pom file is preserved as the parent of projects generated from the archetype?

File: org.eclipse.ice.archetypes/src/main/java/org/eclipse/ice/App.java (hunk 4 above)
Comment (gecage952) [reply org.eclipse.ice.archetypes/src/main/java/org/eclipse/ice/App.java:13]: Yeah, I should remove that.

File: org.eclipse.ice.archetypes/data-element-archetype/src/main/resources/archetype-resources/pom.xml (hunk 2 above)
Comment (gecage952) [reply org.eclipse.ice.archetypes/data-element-archetype/src/main/resources/archetype-resources/pom.xml:47]: Ok. Noted. I'll remove it and make sure everything is good.

File: org.eclipse.ice.archetypes/src/main/java/org/eclipse/ice/App.java (hunk 4 above)
Comment (gecage952) [reply org.eclipse.ice.archetypes/src/main/java/org/eclipse/ice/App.java:13]: And the AppTest too...

File: org.eclipse.ice.archetypes/data-element-archetype/src/main/resources/archetype-resources/pom.xml (hunk 1 above)
Comment (jayjaybillings) [reply org.eclipse.ice.archetypes/data-element-archetype/src/main/resources/archetype-resources/pom.xml:29]: Yes, we do.

File: org.eclipse.ice.archetypes/data-element-archetype/src/main/resources/archetype-resources/pom.xml (hunk 2 above)
Comment (jayjaybillings) [reply org.eclipse.ice.archetypes/data-element-archetype/src/main/resources/archetype-resources/pom.xml:47]: Yes

File: org.eclipse.ice.archetypes/data-element-archetype/src/main/resources/archetype-resources/src/main/java/App.java (hunk 3 above)
Comment (jayjaybillings) [reply org.eclipse.ice.archetypes/data-element-archetype/src/main/resources/archetype-resources/src/main/java/App.java:11]: I agree, and furthermore I think it should include a DataElement and a DataModel. We can think about whether or not there should be a bare bones generator and one that just adds the examples, but we should definitely have them.

File: org.eclipse.ice.archetypes/data-element-archetype/src/main/resources/archetype-resources/pom.xml (hunk 1 above)
Comment (gecage952) [reply org.eclipse.ice.archetypes/data-element-archetype/src/main/resources/archetype-resources/pom.xml:29]: The top level pom for the archetypes is set as the parent, which is where I put the annotations dependency. I just did that so, if that dependency needed to be changed, you could change it in one place and every archetype would just pull from that, but maybe it would be better to just put it inside the actual pom template.

File: org.eclipse.ice.archetypes/data-element-archetype/src/main/resources/archetype-resources/src/main/java/App.java (hunk 3 above)
Comment (gecage952) [reply org.eclipse.ice.archetypes/data-element-archetype/src/main/resources/archetype-resources/src/main/java/App.java:11]: Ok. I'll create some general classes along these lines.

**Comment by gecage952** [comment]
> Thanks @osbornjd. Daniel jumped in to help @gecage952 and they got it sorted.
> 
> Since the checks are passing, this PR is open for review and I encourage everyone to look at it. This will be the basis of how we generate projects and code going forward.
> 
> @gecage952 Do you have an example posted anywhere of what it produces?

I'll incorporate some of these changes and then just make a separate repo where I can push some projects generated from the archetypes and post the link, just so you can see.

File: org.eclipse.ice.archetypes/data-element-archetype/pom.xml [hunk 5]
```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <artifactId>org.eclipse.ice.archetypes</artifactId>
```
Comment (jayjaybillings) [inline org.eclipse.ice.archetypes/data-element-archetype/pom.xml:5]: The version should be 3.0-SNAPSHOT

File: org.eclipse.ice.archetypes/data-element-archetype/pom.xml [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    <groupId>org.eclipse.ice</groupId>
+    <version>1.0-SNAPSHOT</version>
+  </parent>
+
+  <groupId>org.eclipse.ice</groupId>
+  <artifactId>data-element-archetype</artifactId>
+  <version>1.0-SNAPSHOT</version>
```
Comment (jayjaybillings) [inline org.eclipse.ice.archetypes/data-element-archetype/pom.xml:12]: 3.0-SNAPSHOT

File: org.eclipse.ice.archetypes/data-element-archetype/pom.xml [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+  <packaging>maven-archetype</packaging>
+
+  <name>Archetype - data-element-archetype</name>
+
+  <!-- FIXME change it to the project's website -->
+  <url>http://www.example.com</url>
+
+  <properties>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <maven.compiler.source>1.7</maven.compiler.source>
```
Comment (jayjaybillings) [inline org.eclipse.ice.archetypes/data-element-archetype/pom.xml:22]: Can we use Java 11 instead?

File: org.eclipse.ice.archetypes/data-element-webform-archetype/src/main/resources/archetype-resources/pom.xml [hunk 8]
```diff
@@ -0,0 +1,221 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	
+	<parent>
+		<groupId>org.eclipse.ice</groupId>
+		<artifactId>data-renderer-starter</artifactId>
+		<version>1.0-SNAPSHOT</version>
```
Comment (jayjaybillings) [inline org.eclipse.ice.archetypes/data-element-webform-archetype/src/main/resources/archetype-resources/pom.xml:10]: Needs to be 3.0-SNAPSHOT
